### DISCUSSION TITLE
How I change the style in google colab?

### DISCUSSION THREAD

**Discussion by Ppaja** [JCBrouwer/maua-stylegan2#18] [Q&A]
I want another set from https://github.com/justinpinkney/awesome-pretrained-stylegan2

How I do it in google colab? 
Problem is im from germany and new to this and its hard to understand what to do, if youre not native speak english

**Comment by JCBrouwer** [comment]
You need to download it to colab somehow. One way would to just be to download the .pkl file to your computer and then drag it to the file window on the left in colab.

Then you need to make sure that you update the file paths in the code to point to that model file instead of the file it downloads and uses by default.

You'll need to change the cell with `python convert_weight.py ...` (to make sure the .pkl file gets converted to .pt) and then also the final cell that renders the video if you change the name of the .pt file that conert_weight.py outputs.

I think by default the file is called freaGAN, so you could ctrl + f for "frea" and get each of the spots you need to change the file name.

Ich hoffe, dass es dir hilft!

**Reply by Ppaja** [reply]
I found another way, but it works not every time. 
i downloaded the .pkl File To my computer and uploaded it to my google drive. Then I renamed the file to „ freaGAN“. 
Because the colab has the drive import as standard way to download a .pkl File. 

With this i was able to use the .pkl File (with the unicorns) 
But then I tried 2 more .pkl files and they don’t work with this method 
when I try to do the next step of colab it gives me an error At the „convert tensorflow .pkl checkpoint to pytorch.pt checkpoint“ I get the error „ ValueError: Shape mismatch: torch.Size([512, 1024]) vs torch.Size([512, 512])“ 

Also there is the Note „ --repo argument must coincide with the repository that the .pkl was trained with“ 
So I guess that’s the problem. But how I know with what this is trained with? 



EDIT: 
I found a solution. But I still have 2 questions; 
1. how I use the function to separate my audio into bass, melodie and so on? 
2. how I make the time between two pictures on the video shorter? Right now the same image is there for long time, but I want it faster. How I do that? 

And thanks for your help :)

**Reply by JCBrouwer** [reply]
Ahh yeah, just renaming your file to the one that the colab is looks for by default works as well.

The convert_weight.py is only written convert the default settings of stylegan2. So if someone trained the model with different settings then it will run into issues. You need to know what settings the model is that you're converting, otherwise it won't have the right setup.

Two settings that are a bit easier to change are the n_mlp (a couple of lines up) and number of latents (512 on this line): https://github.com/rosinality/stylegan2-pytorch/blob/master/convert_weight.py#L246 . In colab you can open the file and change those 2 settings. I can't really tell what's causing the shape mismatch just from the single line you posted, but maybe changing the 512 to 1024 is all it'll take.

The --repo argument is for specifying if it's from the ADA repo or regular stylegan2 repo.

To split your audio you need to install spleeter (`pip install spleeter`) and then run it, you can find more info [here](https://github.com/deezer/spleeter)

By default the video is generated with the chroma latent strategy which will use the same latent whenever the same note is being played. This can make it stay on one image for a long time if the song isn't changing notes often then neither will the image.

One way you could change this is to use the [`ar.spline_loops()`](https://github.com/JCBrouwer/maua-stylegan2/blob/master/audioreactive/latent.py#L85) function. You can replace these two lines in `get_latents()`:
```
    chroma_latents = ar.chroma_weight_latents(chroma, selection)
    latents = ar.gaussian_filter(chroma_latents, 4)
```
with something like:
```
    latents = spline_loops(selection, args.n_frames, n_loops=20)
```
This will loop through the latent vectors in selection 20 times over the course of your video. If you want it faster you can just set n_loops higher.
